pulp_smash/tests/rpm/api_v2/test_noop.py

```diff
@@ -0,0 +1,264 @@
+# coding=utf-8
+"""Test re-publish repository after unassociating content.
+
+Following steps are executed in order to test correct functionality of
+repository created with valid feed.
+
+1. Create repository foo with valid feed, run sync, add distributor to it and
+   publish over http and https.
+2. Pick a unit X and and assert it is accessible.
+3. Remove unit X from repository foo and re-publish.
+4. Assert unit X is not accessible.
+"""
+from __future__ import unicode_literals
+
+import unittest2
+
+from pulp_smash import api, utils, selectors
+from pulp_smash.compat import urljoin
+from pulp_smash.constants import REPOSITORY_PATH, RPM_FEED_URL
+from pulp_smash.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+    xml_handler,
+    NAMESPACE
+)
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+_PUBLISH_DIR = 'pulp/repos/'
+
+
+def get_repo_md(client, distributor, ):
+    """Return parsed repomd.xml file."""
+    client.response_handler = xml_handler
+    path = urljoin('/pulp/repos/', distributor['config']['relative_url'])
+    path = urljoin(path, 'repodata/repomd.xml')
+    root_elem = client.get(path)
+    client.response_handler = api.json_handler
```

Comment:
This function both returns a parsed `repomd.xml` file and also changes `client.response_handler`. This is valid, but it produces tightly coupled code. Changing this function to have the following signature means that more objects are instantiated, but it makes those objects easier to reason about:

```python
def get_repo_md(server_config, distributor):
```